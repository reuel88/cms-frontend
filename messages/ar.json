{
  "HomePage": {
    "title": "next-intl مثال على",
    "description": "هذا مثال أساسي يوضح استخدام <code>next-intl</code> مع جهاز توجيه التطبيقات Next.js. حاول تغيير الإعدادات المحلية في الزاوية العلوية اليسرى وشاهد كيف يتغير المحتوى."
  },
  "AboutPage": {
    "title": "About",
    "description": "<p>The routing is internationalized too.</p><p>If you're using the default language English, you'll see <code>/about</code> in the browser address bar on this page.</p><p>If you change the locale to German, the URL is prefixed with the locale (<code>/de/about</code>).</p>"
  },
  "Error": {
    "title": "Something went wrong!",
    "description": "<p>We've unfortunately encountered an error.</p><p>You can try to <retry>reload the page</retry> you were visiting.</p>"
  },
  "NotFoundPage": {
    "title": "Page not found",
    "description": "Please double-check the browser address bar or use the navigation to go to a known page."
  },
  "LocaleLayout": {
    "title": "next-intl مثال على"
  },
  "LocaleSwitcher": {
    "label": "Change language",
    "locale": "{locale, select, de {German} en {English} ar {Arabic} other {Unknown}}"
  },
  "Navigation": {
    "home": "Home",
    "about": "About"
  },
  "PageLayout": {
    "links": {
      "docs": {
        "title": "Docs",
        "description": "Learn more about next-intl in the official docs.",
        "href": "https://next-intl-docs.vercel.app/"
      },
      "source": {
        "title": "Source code",
        "description": "Browse the source code of this example on GitHub.",
        "href": "https://github.com/amannn/next-intl/tree/main/examples/example-next-13"
      }
    }
  }
}