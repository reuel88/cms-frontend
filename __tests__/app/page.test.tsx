import { render, screen } from "@testing-library/react";
import Home from "@/app/[locale]/(page)/(routes)/page";
import { NextIntlClientProvider } from "next-intl";
import messages from "../../messages/en.json";

jest.mock("next-intl/client", () => {
  return {
    usePathname: () => "/",
    useRouter: jest.fn(),
  };
});

describe("Home", () => {
  it("renders correctly", () => {
    render(
      <NextIntlClientProvider locale="en" messages={messages}>
        <Home />
      </NextIntlClientProvider>,
    );

    const heading = screen.getByText(/Get started by editing/i);

    expect(heading).toBeInTheDocument();
  });

  it("renders homepage unchanged", () => {
    const { container } = render(
      <NextIntlClientProvider locale="en" messages={messages}>
        <Home />
      </NextIntlClientProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
