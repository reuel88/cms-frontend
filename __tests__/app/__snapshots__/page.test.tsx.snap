// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-between p-24"
  >
    <div
      class="z-10 w-full max-w-5xl items-center justify-between font-mono text-sm lg:flex"
    >
      <p
        class="fixed left-0 top-0 flex w-full justify-center border-b border-gray-300 bg-gradient-to-b from-zinc-200 pb-6 pt-8 backdrop-blur-2xl dark:border-neutral-800 dark:bg-zinc-800/30 dark:from-inherit lg:static lg:w-auto  lg:rounded-xl lg:border lg:bg-gray-200 lg:p-4 lg:dark:bg-zinc-800/30"
      >
        Get started by editing 
        <code
          class="font-mono font-bold"
        >
          src/app/page.tsx
        </code>
      </p>
      <div
        class="fixed bottom-0 left-0 flex h-48 w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          <svg
            class="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <path
              d="M12 2v2"
            />
            <path
              d="M12 20v2"
            />
            <path
              d="m4.93 4.93 1.41 1.41"
            />
            <path
              d="m17.66 17.66 1.41 1.41"
            />
            <path
              d="M2 12h2"
            />
            <path
              d="M20 12h2"
            />
            <path
              d="m6.34 17.66-1.41 1.41"
            />
            <path
              d="m19.07 4.93-1.41 1.41"
            />
          </svg>
          <svg
            class="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
            />
          </svg>
          <span
            class="sr-only"
          >
            Toggle theme
          </span>
        </button>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
          data-state="closed"
          id="radix-:r6:"
          type="button"
        >
          English
          <span
            class="sr-only"
          >
            Toggle Language
          </span>
        </button>
      </div>
    </div>
    <div
      class=""
    >
      <h1
        class="mb-4 text-3xl font-semibold leading-tight tracking-tight text-primary md:text-5xl"
      >
        next-intl example
      </h1>
      <p
        class="max-w-[590px]"
      >
        This is a basic example that demonstrates the usage of 
        <code
          class="font-mono text-primary"
        >
          next-intl
        </code>
         with the Next.js App Router. Try changing the locale in the top right corner and see how the content changes.
      </p>
    </div>
    <div
      class="mb-32 grid text-center lg:mb-0 lg:grid-cols-5 lg:text-left"
    >
      <a
        class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
        href="/about"
      >
        <h2
          class="mb-3 text-2xl font-semibold"
        >
          About
           
          <span
            class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
          >
            -&gt;
          </span>
        </h2>
        <p
          class="m-0 max-w-[30ch] text-sm opacity-50"
        >
          About page.
        </p>
      </a>
      <a
        class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
        href="https://nextjs.org/docs?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          class="mb-3 text-2xl font-semibold"
        >
          Docs
           
          <span
            class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
          >
            -&gt;
          </span>
        </h2>
        <p
          class="m-0 max-w-[30ch] text-sm opacity-50"
        >
          Find in-depth information about Next.js features and API.
        </p>
      </a>
      <a
        class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
        href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template-tw&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          class="mb-3 text-2xl font-semibold"
        >
          Learn
           
          <span
            class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
          >
            -&gt;
          </span>
        </h2>
        <p
          class="m-0 max-w-[30ch] text-sm opacity-50"
        >
          Learn about Next.js in an interactive course with quizzes!
        </p>
      </a>
      <a
        class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
        href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          class="mb-3 text-2xl font-semibold"
        >
          Templates
           
          <span
            class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
          >
            -&gt;
          </span>
        </h2>
        <p
          class="m-0 max-w-[30ch] text-sm opacity-50"
        >
          Explore the Next.js 13 playground.
        </p>
      </a>
      <a
        class="group rounded-lg border border-transparent px-5 py-4 transition-colors hover:border-gray-300 hover:bg-gray-100 hover:dark:border-neutral-700 hover:dark:bg-neutral-800/30"
        href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        <h2
          class="mb-3 text-2xl font-semibold"
        >
          Deploy
           
          <span
            class="inline-block transition-transform group-hover:translate-x-1 motion-reduce:transform-none"
          >
            -&gt;
          </span>
        </h2>
        <p
          class="m-0 max-w-[30ch] text-sm opacity-50"
        >
          Instantly deploy your Next.js site to a shareable URL with Vercel.
        </p>
      </a>
    </div>
  </main>
</div>
`;
